large-message-core/src/main/java/com/bakdata/kafka/AbstractLargeMessageConfig.java

```diff
@@ -282,18 +308,44 @@ private boolean enableAmazonS3PathStyleAccess() {
 
             if (!isEmpty(jwtPath)) {
                 final AWSCredentialsProvider oidcProvider = WebIdentityTokenCredentialsProvider.builder()
-                                .webIdentityTokenFile(jwtPath)
-                                .roleArn(roleArn)
-                                .roleSessionName(roleSessionName)
-                                .build();
+                        .webIdentityTokenFile(jwtPath)
+                        .roleArn(roleArn)
+                        .roleSessionName(roleSessionName)
+                        .build();
 
                 return Optional.of(oidcProvider);
             }
 
-
         }
 
         return Optional.empty();
     }
 
+    /**
+     * This method builds the Google Storage Client object. If you don't specify credentials when constructing the
+     * client, the client library will look for credentials via the environment variable GOOGLE_APPLICATION_CREDENTIALS.
+     * If the environment variable GOOGLE_APPLICATION_CREDENTIALS isn't set,  Application Default Credentials (ADC) uses
+     * the service account that is attached to the resource that is running your code. For more information see the <a
+     * href="https://cloud.google.com/docs/authentication/production#automatically">official documentation</a>
+     *
+     * @return GoogleStorageClient
+     */
+    private BlobStorageClient createGoogleStorageClient() {
+        if (!this.getString(GOOGLE_CLOUD_KEY_PATH).isEmpty()) {
+            final GoogleCredentials credentials;
+            try {
+                credentials = GoogleCredentials.fromStream(new FileInputStream(GOOGLE_CLOUD_KEY_PATH))
```

Comment:
The fileinputstream should be closed using a tray-catch-with-resources-block. Also I would move the whole try to a method `getGoogleCredentials` so that you don't have to define the variable outside a try block